framework/src/userobject/ActivateElementsCoupled.C

```diff
@@ -0,0 +1,64 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ActivateElementsCoupled.h"
+
+registerMooseObject("MooseApp", ActivateElementsCoupled);
+
+InputParameters
+ActivateElementsCoupled::validParams()
+{
+  InputParameters params = ActivateElementsUserObjectBase::validParams();
+
+  params.addRequiredParam<Real>("activate_value", "The value above which to activate the element");
+  params.addCoupledVar(
+      "coupled_var",
+      "The variable value will be used to decide wether an element whould be activated.");
+  params.addParam<MooseEnum>("activate_type",
+                             MooseEnum("below equal above", "above"),
+                             "Activate element when below or above the activate_value");
+  return params;
+}
+
+ActivateElementsCoupled::ActivateElementsCoupled(const InputParameters & parameters)
+  : ActivateElementsUserObjectBase(parameters),
+    _coupled_var(isParamValid("coupled_var") ? &coupledValue("coupled_var") : nullptr),
+    _activate_value(
+        declareRestartableData<Real>("activate_value", getParam<Real>("activate_value"))),
+    _activate_type(getParam<MooseEnum>("activate_type").getEnum<ActivateType>())
+{
+}
+
+bool
+ActivateElementsCoupled::isElementActivated()
+{
+  bool is_activated = false;
+  Real avg_val = 0.0;
+
+  for (unsigned int qp = 0; qp < _qrule->n_points(); ++qp)
+    avg_val += (*_coupled_var)[qp];
```

Comment:
Looks like `_coupled_var` could be `nullptr`

Reply:
Good point! Can we make `_coupled_var` a required parameter? Otherwise I may just throw `paramError` when it is `nullptr`

Reply:
Absolutely. I just realized that both this and the functions were only pointers because you had them in the same class before. Make it required and use reference.

Reply:
Thanks @dschwen ! Just to make sure, by this you mean something like:
`const VariableValue _coupled_var;` in the header,
`params.addRequiredCoupledVar("coupled_var", "  ");` in the `ActivateElementsCoupled::validParams()`, and 
`_coupled_var(coupledValue("coupled_var"))` in the constructor?

The function paths maybe 1, 2 or 3D for different cases so it maybe better to not make them required, I guess?